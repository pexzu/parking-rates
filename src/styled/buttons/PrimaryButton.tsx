import styled from '@emotion/styled';

export const PrimaryButton = styled('button')`
  label: PrimaryButton;

  padding: 6px 15px;
  margin: 6px;
  color: #ffffff;
  background: #224172;
  border-radius: 3px;
  border: 0;
  transition: 0.3s ease;
  cursor: pointer;

  &:hover {
    opacity: 0.9;
  }
  &:disabled {
    background: #d3d3d3;
    cursor: not-allowed;
    &:hover {
      opacity: 1;
    }
  }
`;
