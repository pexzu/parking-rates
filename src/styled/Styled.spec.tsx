import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import { PrimaryButton } from './buttons/PrimaryButton';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('PrimaryButton component', () => {
  test('render', () => {
    const wrapper = shallow(<PrimaryButton />);
    expect(wrapper.exists()).toBe(true);
  });
});

//Snapshot Testing
it('PrimaryButton renders correctly', () => {
  const tree = renderer.create(<PrimaryButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
