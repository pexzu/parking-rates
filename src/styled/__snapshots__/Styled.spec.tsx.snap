// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton renders correctly 1`] = `
<button
  className="css-ofp0oc-PrimaryButton"
/>
`;
